Source/ARTDefault.m

```diff
@@ -135,4 +165,64 @@ + (NSString *)bundleBuildNumber {
     return infoDictionary[ARTDefault_bundleBuildNumberKey];
 }
 
++ (NSString *)osName {
+    return
+        #if TARGET_OS_IOS
+            @"iOS"
+        #elif TARGET_OS_TV
+            @"tvOS"
+        #elif TARGET_OS_WATCH
+            @"watchOS"
+        #elif TARGET_OS_OSX
+            @"macOS"
+        #else
+            nil
+        #endif
+        ;
+}
+
++ (NSString *)osVersionString {
+    static NSString *versionString;
+    static dispatch_once_t onceToken;
+    dispatch_once(&onceToken, ^{
+        NSOperatingSystemVersion version = [[NSProcessInfo processInfo] operatingSystemVersion];
+        versionString = [NSString stringWithFormat:@"%u.%u.%u",
+                         conformVersionComponent(version.majorVersion),
+                         conformVersionComponent(version.minorVersion),
+                         conformVersionComponent(version.patchVersion)];
+    });
+    return versionString;
+}
+
++ (NSString *)deviceModel {
+    struct utsname systemInfo;
+    if (uname(&systemInfo) < 0) {
+        return nil;
+    }
+    return [NSString stringWithCString:systemInfo.machine encoding:NSUTF8StringEncoding];
+}
+
++ (NSString *)deviceModelToken {
+    static NSString *modelToken;
+    static dispatch_once_t onceToken;
+    dispatch_once(&onceToken, ^{
+        NSString *deviceModel = [self deviceModel];
+        modelToken = deviceModel != nil ? canonizeStringAsAgentToken(deviceModel) : nil;
+    });
+    return modelToken;
+}
+
++ (NSString *)agent {
+    NSMutableString *agentString = [NSMutableString stringWithFormat:@"%@/%@", ARTDefault_libName, [self bundleVersion]];
+    NSString *osName = [self osName];
+    if (osName != nil) {
+        [agentString appendFormat:@" %@/%@", osName, [self osVersionString]];
+    }
+    NSString *deviceModelToken = [self deviceModelToken];
+    if (deviceModelToken != nil) {
+        [agentString appendFormat:@" %@", deviceModelToken];
```

Comment:
For the time being we will drop any attempt to collect device information. The above `osName`/`osVersionString` is adequate for our needs, and matches [the requirement](https://docs.ably.com/client-lib-development-guide/features/#RSC7d5) to have all possible product string emissions databased in [agents.json](https://github.com/ably/ably-common/blob/main/protocol/agents.json).

Reply:
https://github.com/ably/ably-cocoa/pull/1119/commits/088bec2fc18adf12ef8528fd2f39f3da758c005d